subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestExecutionListener.java

```diff
@@ -163,7 +161,7 @@ private boolean createDescriptorIfAbsent(TestIdentifier node) {
             if (node.getType() == CONTAINER || isTestClassIdentifier(node)) {
                 TestIdentifier classIdentifier = findTestClassIdentifier(node);
                 String className = className(classIdentifier);
-                String classDisplayName = classDisplayName(classIdentifier);
+                String classDisplayName = node.getDisplayName();
                 return new DefaultTestClassDescriptor(idGenerator.generateId(), className, classDisplayName);
```

Comment:
❌ Using `DefaultTestClassDescriptor` for these intermediate items makes the progress logger problem worse since all of them will now be shown as running in parallel because it does an `instanceof` check: https://github.com/gradle/gradle/blob/e753a15d49d8ad5d311055e470812389a3ce8f7a/subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/TestWorkerProgressListener.java#L90-L98

I think `TestWorkerProgressListener` should be changed to only show the topmost `DefaultTestClassDescriptor` (that would also solve the problem for `@Nested` tests which are, technically, test classes).